### PR TITLE
Improve matching custom `java.lang.annotation.Annotation`

### LINKED ISSUE #270: `java.lang.reflect.Type`s equality should't be asserted by `equals` for some cases
`org.jboss.cdi.tck.tests.event.metadata.EventMetadataTest`

Micronaut will return a custom instance of `ParameterizedType` which is not going to match the reflection implementation.

### PR DESCRIPTION
**Pull request by dstepanov**
Fixes #270

### CODE REVIEW

File: impl/src/main/java/org/jboss/cdi/tck/util/Assert.java [hunk 1]
```diff
@@ -96,9 +108,49 @@ public static void assertTypeListMatches(List<? extends Type> types, Type... req
...
+        for (Type existingType : existingTypes) {
+            if (isTypesEqual(requiredType, existingType)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static Boolean isTypesEqual(Type type, Type requiredType) {
+        return type.getTypeName().equals(requiredType.getTypeName());
```
Comment (Ladicek) [inline impl/src/main/java/org/jboss/cdi/tck/util/Assert.java:141]: This seems rather imprecise? Also `Type.getTypeName()` by defaults delegates to `toString()`, so it's also quite fragile in presence of custom implementations of `Type`. IIUC, the original code relied on `Type.equals()`, which is likely to be much better. Of course it requires custom implementations of `Type` to interoperate correctly, which isn't that hard to do AFAICT. Here are some custom implementations of `Type` that (should :-) ) work correctly: https://github.com/weld/core/tree/master/weld-lite-extension-translator/src/main/java/org/jboss/weld/lite/extension/translator/util/reflection

Comment (Ladicek) [reply impl/src/main/java/org/jboss/cdi/tck/util/Assert.java:141]: (It also requires the JDK implementations of `Type` to interoperate correctly, which AFAIK they do.)

Comment (dstepanov) [reply impl/src/main/java/org/jboss/cdi/tck/util/Assert.java]: You are right, I wound a bug in our type implementation. Keeping only annotations changes.

**Review by manovotn — DISMISSED** [review]
I am fine with these changes but leaving the merge to @Ladicek as he had some comments to be addressed.

**Comment by Ladicek** [comment]
No, my previous comment was addressed. But I keep wondering why this is necessary. Are you perhaps generating annotation classes that don't implement `annotationType()` properly?

**Comment by dstepanov** [comment]
Good question, I will investigate more :)

**Comment by manovotn** [comment]
@dstepanov is there any update on this PR? Should we keep it open?

**Comment by dstepanov** [comment]
Not sure if it is still needed, we can close it for now

**Pull request closed without merging**